src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java

```diff
@@ -30,102 +30,414 @@
+    }
+    
+    /**
+     * Finish test item for the specified extension context
+     * 
+     * @param context extension context
+     */
+    private void finishTestItem(ExtensionContext context) {
+        // instantiate "finish test item" request
+        FinishTestItemRQ rq = new FinishTestItemRQ();
+        // set test item status
+        rq.setStatus(getExecutionStatus(context));
+        // set test item end time
+        rq.setEndTime(Calendar.getInstance().getTime());
+        // finish test item for extension context, remove mapping [context => test item]
+        getLaunchFor(context).finishTestItem(idMapping.remove(context.getUniqueId()), rq);
+    }
+    
+    /**
+     * Get execution status for the specified extension context.
+     * 
+     * @param context extension context
+     * @return {@link Statuses status constant}
+     */
+    private static String getExecutionStatus(ExtensionContext context) {
+        Optional<Throwable> exception = context.getExecutionException();
+        if (!exception.isPresent()) {
+            return Statuses.PASSED;
+        } else if (exception.get() instanceof TestAbortedException) {
+            return Statuses.SKIPPED;
+        } else {
+            return Statuses.FAILED;
+        }
+    }
+    
+    /**
+     * If not already done, start a containing suite for the specified template test invocation.
+     * 
+     * @param context extension context for template test invocation
+     * @return test item ID for containing suite (may be {@code empty})
+     */
+    private synchronized Maybe<String> startSuiteIfRequiredFor(ExtensionContext context) {
+        TemplateTestSuite testSuite;
+        Maybe<String> suiteId = Maybe.empty();
+        // if template test invocation
+        if (isTemplateTestInvocation(context)) {
+            // get unique ID of context parent
+            String parentId = context.getParent().get().getUniqueId();
+            // if containing suite already started
+            if (templateTestSuites.containsKey(parentId)) {
+                // get containing template test suite object
+                testSuite = templateTestSuites.get(parentId);
+            // otherwise (suite not yet started)
+            } else {
+                // start containing template test suite
+                testSuite = TemplateTestSuite.startTestSuiteFor(context, this);
+                // store association: parent context => suite
+                templateTestSuites.put(parentId, testSuite);
+            }
+            // get test item ID for suite
+            suiteId = testSuite.getSuiteId();
+        }
+        return suiteId;
+    }
+    
+    /**
+     * Get containing suite for the specified template test invocation.
+     * 
+     * @param context extension context for template test invocation
+     * @return containing template test suite object (may be {@code null})
+     */
+    private synchronized TemplateTestSuite getTestSuiteFor(ExtensionContext context) {

```

Comment:
there is no need to mark it as synchronized

Reply:
I removed the `synchronized` keyword from:
* startSuiteIfRequiredFor()
* getTestSuiteFor()
* finishTestSuiteFor()